CMakeLists.txt

```diff
@@ -77,9 +77,29 @@ endif()
 FetchContent_MakeAvailable(SwiftFoundationICU SwiftFoundation)
 
 include(CheckLinkerFlag)
+include(CheckSymbolExists)
 
 check_linker_flag(C "LINKER:--build-id=sha1" LINKER_SUPPORTS_BUILD_ID)
 
+check_symbol_exists("strlcat" "string.h" HAVE_STRLCAT)
+check_symbol_exists("strlcpy" "string.h" HAVE_STRLCPY)
+check_symbol_exists("issetugid" "unistd.h" HAVE_ISSETUGID)
+add_compile_definitions(
+  $<$<AND:$<COMPILE_LANGUAGE:C>,$<BOOL:${HAVE_STRLCAT}>>:HAVE_STRLCAT>
+  $<$<AND:$<COMPILE_LANGUAGE:C>,$<BOOL:${HAVE_STRLCPY}>>:HAVE_STRLCPY>
+  $<$<AND:$<COMPILE_LANGUAGE:C>,$<BOOL:${HAVE_ISSETUGID}>>:HAVE_ISSETUGID>)
```

Comment:
We've discussed these before a handful of times (see https://github.com/apple/swift-corelibs-foundation/pull/5013) but SwiftPM does not support these types of definitions so this is incompatible with the SwiftPM build. Instead I believe we need to conditionally define these in the headers/implementation files rather than in the CMake/package manifest files

Reply:
SPM will need to learn how to query what is available in the SDK then...
From having worked with internal vs external SDKs, there are certain Linux distros (that I have to support) where certain symbols are only exposed if you're working internally. The version numbers, outward names, and other macro definitions are identical in both cases, but the available functions aren't the same, so trying to hard-code a mish-mash of definitions into the source file statically isn't going to work.

Reply:
I agree with the sentiment and definitely understand that the workaround is not a long term solution. This is definitely an area where SwiftPM should grow to support these needs.

For the time being since we don't have that capability in SwiftPM yet, are we able to work around that limitation for this case specifically? SwiftPM builds are one of our main workflows for developing on this project, and is how we run tests for this project as well so unfortunately neither adding a new SwiftPM feature nor breaking the SwiftPM build is an option. But if we have a way to get these features to work specifically (or a way to detect this for only the static build since SwiftPM doesn't build the Static SDK) then that seems like a workable solution until SwiftPM is capable of doing this

Reply:
The checks for the macros are already in this codebase, this just added the check back to CMake, so as far as SPM is concerned, it will continue to behave exactly as it is today. This just means that folks who are building for systems that define these functions in their libc have a way to build corelibs-foundation as well without having duplicate-definitions.

Reply:
Ok, as it stands today, `HAVE_STRLCAT` and `HAVE_STRLCPY` are defined to `1` on WASI via:

```c
#if TARGET_OS_WASI
#define HAVE_STRLCPY 1
#define HAVE_STRLCAT 1
#endif
```

so it sounds like this is for the Static SDK based on Musl - since we don't currently expect to be able to build the Static SDK with SwiftPM, can you add a comment here that makes it explicitly noted that this is to support building against Musl for the Static SDK and therefore does not need to be replicated in the SwiftPM package manifest, just to avoid future confusion?